contracts/cw721-base/src/execute.rs

```diff
@@ -14,16 +16,17 @@ use crate::state::{Approval, Cw721Contract, TokenInfo};
 const CONTRACT_NAME: &str = "crates.io:cw721-base";
 const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
 
-impl<'a, T, C, E, Q> Cw721Contract<'a, T, C, E, Q>
+impl<'a, T, E1, E2, C, Q> Cw721Contract<'a, T, E1, E2, C, Q>
 where
     T: Serialize + DeserializeOwned + Clone,
+    E1: DeserializeOwned,
+    E2: DeserializeOwned,
     C: CustomMsg,
-    E: CustomMsg,
-    Q: CustomMsg,
+    Q: CustomQuery,
```

Comment:
This is going to need to get rebased on this change which also changes these generic names: https://github.com/CosmWasm/cw-nfts/pull/75

How would you feel about giving `Q` here a more descriptive name? `DepsQueryExt`?

Reply:
I would name `Q` into `ModuleQuery` and `C` into `ModuleMsg`. What do you think?